enginetest/spatial_index_tests.go

```diff
@@ -111,7 +111,7 @@ var SpatialIndexTests = []SpatialIndexPlanTest{
 				},
 			},
 			{
-				noIdx: true,
+				noIdx: false, // still expect index access using primary key
```

Comment:
I'm not sure how this incorrect test made it through the gaps; it is likely due to the bad `replacePkSort` rule.
This should do the same thing as the test right about.